ls1intum/Artemis
# `Development`: Fix e2e quiz exercise drop location test failing

### Checklist
#### General


#### Server

### Motivation and Context
"Quiz exercise drop location" e2e test fails sometimes due to:
1. Quiz preview not being loaded
2. Generate quiz button not being loaded/available when clicked
3. Course exercises page is visited before new exercise is generated

### Description
1. Added assertion for quiz preview to become visible before doing preview related assertions
2. Added assertion for generate button to be visible, followed by short delay
3. Added assertion for quiz details to be shown after generate button is clicked, to make sure it successfully finished generation

### Steps for Testing

Run **QuizExerciseDropLocation** test on cypress several times to check if it consistently passes.

### Review Progress

#### Code Review
#### Manual Tests
